lib/cog/chat/slack/formatter.ex

```diff
@@ -62,11 +62,26 @@ defmodule Cog.Chat.Slack.Formatter do
       nil ->
         message
       [{start, length}|_] ->
-        original = String.slice(message, start, length)
+        # Regex.run returns indices of a char list which causes weirdness when
+        # unicode is present. We have to convert with erlang's older binary to
+        # list conversion (ignoring unicode codepoints), do the replace, and
+        # then convert back with the same older conversion function for the
+        # indices to match up correctly. This is possibly a bug in Elixir's
+        # `Regex.run`.
```

Comment:
Is this because `{start, length}` are byte indexes instead of Unicode code points? If so, are there any guarantees they're aligned to Unicode character boundaries so we avoid generating invalid strings?

I was looking at this code last night and wondered why we're not using named capture groups to extract the desired value and `String.replace` to edit the string. It would require running regexes twice but avoid having to care about string indexing at all.